public/assets/js/index.js

```diff
@@ -0,0 +1,314 @@
+
+    if (data.success) {
+      window.location.replace("/dashboard");
+    } else {
+      const warning = `<div class="alert alert-warning" role="alert">
+   Incorrect Email or Password 
+   </div>`;
+      alertMessage.empty();
+
+      return alertMessage.append(warning);
+    }
+  } else {
+    const warning = `<div class="alert alert-warning" role="alert">
+  Please fill in the fields to log in
+ </div>`;
+    alertMessage.empty();
+
+    return alertMessage.append(warning);
+  }
+};
+
+const handleLogout = async () => {
+  const response = await fetch("/auth/logout", {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json",
+    },
+  });
+
+  const data = await response.json();
+
+  if (data.success) {
+    window.location.replace("/login");
+  }
+};
+
+const handleCreateBlogPost = async (event) => {
+  event.preventDefault();
+
+  const title = $("#blog-title").val();
+  const blog_img = $("#blog-img").val();
+  const content = $("#blog-content").val();
+  const alertMessage = $("#alert-message");
+
+  if (title && blog_img && content) {
+    const response = await fetch("/api/blog", {
+      method: "POST",
+      headers: {
+        Accept: "application/json",
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({
+        title,
+        blog_img,
+        content,
+      }),
+      redirect: "follow",
+    });
+
+    const data = await response.json();
+
+    if (data.success) {
+      window.location.replace("/dashboard");
+    }
+  } else {
+    const warning = `<div class="alert alert-warning" role="alert">
+  Please fill in all the fields
+ </div>`;
+    alertMessage.empty();
+
+    return alertMessage.append(warning);
+  }
+};
+
+const handleEditBlogPost = async (event) => {
+  event.preventDefault();
+
+  // window.location.pathname.split('/')[3]
+
+  const { id } = event.target;
+
+  const title = $("#edit-title").val();
+  const blog_img = $("#edit-img").val();
+  const content = $("#edit-content").val();
+  const alertMessage = $("#alert-message");
+
+  console.log(title, blog_img, content);
+
+  if (title && blog_img && content) {
+    const response = await fetch(`/api/blog/${id}`, {
+      method: "PUT",
+      headers: {
+        Accept: "application/json",
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({
+        title,
+        blog_img,
+        content,
+      }),
+      redirect: "follow",
+    });
+
+    const data = await response.json();
+
+    if (data.success) {
+      window.location.replace("/dashboard");
+    }
+  } else {
+    const warning = `<div class="alert alert-warning" role="alert">
+  Please fill in all the fields 
+ </div>`;
+    alertMessage.empty();
+
+    return alertMessage.append(warning);
+  }
+};
+
+const handleDeleteBlog = async (event) => {
+  event.preventDefault();
+  const { id } = event.target;
+
+  const confirmMessage = confirm("Are you sure you want to delete this blog");
```

Comment:
It is great that you have used validation on the delete buttons to confirm that the user wants to delete a blog/comment. However you shouldn't be using confirm or alerts anymore. Consider using a modal instead to make the application cleaner and more professional.